DEVELOPMENT.md

```diff
@@ -166,15 +166,118 @@ The actual commands for compilation are handled by our Makefile, so the only thi
 │   ├── component       contains the base component; all other components extend this component
 │   ├── config          package for configuration using config files, environment and CLI flags
+│   ├── console         package that provides the web server for the console
 │   ├── errors          package for rich errors that include metadata and cross API boundaries
 │   ├── log             package for logging
 │   ├── messages        contains non-proto messages (such as the messages that are sent over MQTT)
 │   ├── metrics         package for metrics collection
 │   ├── ttnpb           contains generated code from our protocol buffer definitions and some helper functions
 │   ├── types           contains primitive types
+│   ├── webui           contains js code for the console and oauth provider
 │   └── ...
 ├── public              frontend code will be compiled to this folder - not added to git
 ├── release             binaries will be compiled to this folder - not added to git
 └── sdk                 source code for our SDKs
     └── js              source code for our JavaScript SDK
 ```
+
+
+## Frontend
+### Introduction
+The TTN stack for LoRaWAN employs two frontend applications: the **Console** and **OAuth Provider**. Both applications use [React](https://reactjs.org/) as frontend framework. The `console` package of the backend exposes the webserver and – together with the `oauth` package – is in charge of handling authorization logic (OAuth flow, login/logout).
+
+#### Console
+The Console is the official management application of the stack. It can be used to register applications, devices or gateways, to monitor network traffic, or configuring network related options, among other things. It is a single page application (SPA) that runs entirely in the browser and uses an access token to communicate with the stack.
+
+#### OAuth
+The OAuth app provides the necessary frontend for the OAuth provider of the stack. It is used e.g. to display the authorization screen that users get prompted with when they want to authorize a third-party app to access the stack.
+
+### Building the frontend
+You can control whether to build the frontend for production or development by setting the `$NODE_ENV` environment variable to either `development` or `production`. The frontend is then build using:
+
+```sh
+mage js:build
+```
+*Note that the stack needs to be initialized first using `mage init`*
+
+This will initiate the following actions:
+
+* Install a version-fixed binary of `yarn`
+* Install JS SDK node dependencies via `yarn`
+* Build the JS SDK
+* Extract backend locale messages
+* Install frontend node dependencies via `yarn`
+* Build the frontend (using `webpack`) and output into `/public`
+
+The difference of a development build includes:
+
+* Including source maps
+* Using DLL bundle for modules to reduce build time
+* Means to stub backend data to make `webpack-dev-server` work properly
```

Comment:
`webpack-dev-server` is used proxy request from the console to the stack, no stubs.